### DISCUSSION TITLE
v6 update region id

### DISCUSSION THREAD

**Discussion by g-no** [katspaugh/wavesurfer.js#2861] [Q&A] [answered]
When I try to update a regions id I fail.
My region's id's are just consecutive numbers starting with 1. Now I have removed a region (e.g. "6") and want to rename all regions with number/id higher than 6 by -1 like this (e.g. currentId = 7, newId = 6)
```
const region = wavesurfer.regions.list[currentId];
    if (region) {
        console.log("currentId: ", currentId);
        console.log("newId: ", newId);
        console.log("ws id before", region.id);
        region.update({ id: newId });
        console.log("ws id after", region.id);
}
```                    
But I get 
```
currentId:  7
newId:  6 
ws id before 7
ws id after 7
```
Is the `id` property read-only? And do I have to remove and add all regions again?

**Comment by katspaugh** [comment, marked as answer]
You can save each added region into your own array or map and set whatever keys/ids for them.

**Reply by g-no** [reply]
Thanks, this works.
